src/qp/factory.py

```diff
@@ -231,22 +237,22 @@ def iterator(self, filename, chunk_size=100_000, rank=0, parallel_size=1):
         chunk_size : `int`
         """
         extension = os.path.splitext(filename)[1]
-        if extension not in ['.hdf5']:  #pragma: no cover
+        if extension not in [".hdf5"]:  # pragma: no cover
             raise TypeError("Can only use qp.iterator on hdf5 files")
 
-        metadata = io.readHdf5ToDict(filename, 'meta')
-        pdf_name = metadata.pop('pdf_name')[0].decode()
-        pdf_version = metadata.pop('pdf_version')[0]
-        if pdf_name not in self: #pragma: no cover
+        metadata = io.readHdf5ToDict(filename, "meta")
+        pdf_name = metadata.pop("pdf_name")[0].decode()
+        _pdf_version = metadata.pop("pdf_version")[0]
```

Comment:
I take it you want to pop this to maintain *some* behavior, but it's otherwise ignored?

Reply:
_pdf_version is there for if/when we need to migrate schema.   I'd just as soon leave it in, so that that we don't have to go digging around for it when that time comes.   And yes, the `pop` is to make sure it doesn't get sent to functions that aren't expecting it.